distributed/scheduler.py

```diff
@@ -2801,19 +2801,17 @@ def _set_duration_estimate(self, ts: TaskState, ws: WorkerState) -> None:
         if ts in ws.long_running:
             return
 
-        exec_time: float = ws.executing.get(ts, 0)
-        duration: float = self.get_task_duration(ts)
-        total_duration: float
-        if exec_time > 2 * duration:
-            total_duration = 2 * exec_time
-        else:
-            comm: float = self.get_comm_cost(ts, ws)
-            total_duration = duration + comm
+        exec_time = ws.executing.get(ts, 0.0)
+        cpu = self.get_task_duration(ts)
+        if exec_time > 2 * cpu:
+            cpu = 2 * exec_time
+        network = self.get_comm_cost(ts, ws)
 
-        old = ws.processing.get(ts, 0)
-        ws.processing[ts] = total_duration
-        self.total_occupancy += total_duration - old
-        ws.occupancy += total_duration - old
+        old = ws.processing.get(ts, Occupancy(0, 0))
```

Comment:
nit: We're calling this method a lot. This will always initialize the empty class instance

Reply:
Yeah, maybe worth a separate branch for the case that the key isn't there? Or allowing `Occupancy` to be incremented and decremented by plain ints?